### PR TITLE
Two fixes in Raster.reproject

### PR DESCRIPTION
**Pull request by adehecq**
- solves #453 
- fixes an issue with reproject not enforcing bounds in some cases due to floating point precision (see [issue](https://github.com/rasterio/rasterio/issues/3016) raised in rasterio). Extracted window need to be rounded to nearest integer rather than up.

Weirdly the second change does not make any difference for tests. I guess in all tested cases, the rounding does not make any difference. This was probably an edge case that I encountered.
Not sure if there is any test to add for this...

### CODE REVIEW

File: tests/test_raster.py [hunk 1]
```diff
@@ -1381,6 +1381,13 @@ def test_reproject(self, example: str) -> None:
         assert r_test.crs.to_epsg() == 4326
 
         # -- Additional tests --
+        # First, make sure dst_bounds extend beyond current extent to create nodata
```
Comment (rhugonnet) [inline tests/test_raster.py:1384]: So this is a replacement of the test never run? Should we delete that test that is never run then?

**Review by rhugonnet — APPROVED** [review]
If you still have the edge case that you can re-use to test the rounding, it'd be a nice quick addition to the tests.
Otherwise, all good for me!

File: tests/test_raster.py (hunk 1 above)
Comment (adehecq) [reply tests/test_raster.py:1384]: Sorry, there's a mistake I think. `r_test` should be used in the following lines, because I made sure it contains nodata. Before these lines were added, there was actually no gap so the test was pointless...
I'll check again!

Comment (adehecq) [reply tests/test_raster.py:1384]: In the end, no it was correct, `r_test` is just used to check that reproject with these bounds will creates nodata in the following tests, not used directly in those tests.
I had to follow several layers of git blame to track down the origin of this test. It was created [here](https://github.com/GlacioHack/geoutils/pull/223), but then modified a few times, particularly [here](https://github.com/GlacioHack/geoutils/pull/284) by you @rhugonnet 😜, which made the test lose its usefulness...
In short, it was just added at a point where rasterio's default nodata values would add very large values. I'm not sure this test is super useful nowadays, but I will keep it just in case we break something very badly.

**Comment by adehecq** [comment]
> If you still have the edge case that you can re-use to test the rounding, it'd be a nice quick addition to the tests. Otherwise, all good for me!

Unfortunately, I only have partial information in the rasterio issue I raised, so I cannot fully reproduce now. I will ignore it for the moment.

**Comment by rhugonnet** [comment]
Perfect! Also, your tests are passing... While it's still failing on #462 with a segmentation error, even if I reverted back nearly everything, I don't understand what's going on... :exploding_head:

**Comment by rhugonnet** [comment]
Actually I know! Your PR didn't have any environment file change, so it's still using the cached environment that was working...
OK I'll keep investigating in #462
